src/Control/Monad/Trans/OptionParser.hs

```diff
@@ -85,7 +86,7 @@ instance MonadIO m => MonadIO (OptionParserT o m) where
 mapOptionParserT :: (forall a. m a -> m' a)
                  -> OptionParserT o m b -> OptionParserT o m' b
 mapOptionParserT f = OptionParserT
```

Comment:
Yepp, hlint src/Control/Monad/Trans/OptionParser.hs suggests

```log
src/Control/Monad/Trans/OptionParser.hs:88:22: Suggestion: Move brackets to avoid $
Found:
  (mapStateT $ mapStateT $ mapUncertainT f) . unOptionParserT
Why not:
  mapStateT (mapStateT $ mapUncertainT f) . unOptionParserT

src/Control/Monad/Trans/OptionParser.hs:389:31: Warning: Monad law, right identity
Found:
  consumeString input >>= check >>= return
Why not:
  consumeString input >>= check

src/Control/Monad/Trans/OptionParser.hs:511:25: Suggestion: Use <$>
Found:
  fmap reverse $ go []
Why not:
  reverse <$> go []

3 hints
```